torch/utils/checkpoint.py

```diff
@@ -35,6 +35,37 @@ def check_backward_validity(inputs: Iterable[Any]) -> None:
         warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
 
 
+def _get_device_module(device='cuda'):
+    device_module = getattr(torch, device)
+    return device_module
+
+class _DefaultDevice(object):
```

Comment:
@heidongxianhua 

Are we saying that we still want to temporarily provide this API (e.g. to set_device_type), but since its temporary and subject to change, we'd like to keep it non-official/private?

If that is the case I'd rather say that users just shouldn't call this because we plan to remove it once we support all devices, and in that case since I don't see it called anywhere else in this file, maybe we should just hardcode this value to cuda?

Reply:
> @heidongxianhua
> 
> Are we saying that we still want to temporarily provide this API (e.g. to set_device_type), but since its temporary and subject to change, we'd like to keep it non-official/private?
> 
> If that is the case I'd rather say that users just shouldn't call this because we plan to remove it once we support all devices, and in that case since I don't see it called anywhere else in this file, maybe we should just hardcode this value to cuda?

yeah, I have renamed the func to make it as private (by prepending underscore to its name ). And now default value is cuda, and if we set it to be other device (for example, custom-device PrivateUse1), it will work well.

Reply:
What I meant is that if we'd like users to use it (e.g. to update it to be PrivateUse1), wouldn't it be public then? And if so, we would also need some documentation.

Now that I think about it, even if plan to support multiple device types in the future, that wouldn't break this API - it would just make it a no-op, so public is probably fine.

Reply:
> What I meant is that if we'd like users to use it (e.g. to update it to be PrivateUse1), wouldn't it be public then? And if so, we would also need some documentation.
> 
> Now that I think about it, even if plan to support multiple device types in the future, that wouldn't break this API - it would just make it a no-op, so public is probably fine.

sorry to mistake it, @soulitzer 😀yeah，it will be used by user, and it is better to make it as public. And I will give a commit to do this and add document to explain how to use.